norse/torch/functional/lif.py

```diff
@@ -392,8 +429,28 @@ def lif_feed_forward_integral(
         p (LIFParameters): parameters of a leaky integrate and fire neuron
         dt (float): Integration timestep to use
     """
-    z, v, i = norse_op.lif_super_feed_forward_integral(input_tensor, state, p, dt)
-    return z, LIFState(z=z, v=v, i=i)
+    outputs = []
```

Comment:
Why not use the cpp code? If you are concerned about parameterizing the surrogate gradient method, I suggest renaming the function to include `super` and then do the differentiation in the `LIF` module, such so we retain the performance benefits.

Reply:
I don't think special casing like this is a good idea, right now. It breaks the expectation of the users and violates the interface.

Reply:
We already have those "special cases": https://github.com/norse/norse/blob/master/norse/torch/module/lif.py#L65

Reply:
I wouldn't say this is a special case, just a different algorithm. The choice of surrogate gradient function at the moment does nothing *only* in the case of LIF neurons. This is unexpected and not consistent with the documentation.

Reply:
Overall, I think it's desirable to abstract the gradient algorithms in the "method" parameter. The error in LIF neurons, where it's ignored is critically bad. But that doesn't mean we can't still have it. Perhaps I'll finally try to flesh out the Threshold #242  module in the weekend. That could be a good place to start discussing how the API could/should look longer term.

Reply:
But it can't be abstracted like this, what is missing is proper integration support. Similarly for the threshold module, if you want to pass things to C++ you do not want to support necessarily arbitrary callable, which is why I went for the enum change. Baring that the proper way to do it would probably to just pass a callable and eliminate the threshold function altogether.